src/main/java/internet/store/service/GenericService.java

```diff
@@ -0,0 +1,12 @@
+package internet.store.service;
+
+import java.util.List;
+
+public interface GenericService<T, K, M> {
```

Comment:
```suggestion
public interface GenericService<T, K> {
```
you should be able to use only two parameters

Reply:
Roman, I need third for method (Delete) - order and user we delete by id, while Product - by product (as per initial task). I can change method Delete for Product to do delete by id or remove delete from generic - then I'll be able to live with two parameters. What is the best?